packages/evolution-frontend/src/components/inputs/InputRadioNumber.tsx

```diff
@@ -0,0 +1,152 @@
+                    className={'input-radio'}
+                    value={value}
+                    onChange={onChange}
+                    ref={inputRadioRef}
+                />
+                <label htmlFor={id}>
+                    {icon && (
+                        <span>
+                            <FontAwesomeIcon
+                                icon={icon}
+                                className="faIconLeft"
+                                style={{margin: 'auto 0'}}
+                            />
+                        </span>
+                    )}
+                    {inputIconPath && (
+                        <span>
+                            <img
+                                src={inputIconPath.iconPath}
+                                className="faIconLeft"
+                                style={{height: inputIconPath.iconSize, margin: '0 0.25em 0 0'}}
+                                alt={`${value}`}
+                            />
+                        </span>
+                    )}
+                    <span>{value}</span>
+                </label>
+            </div>
+        </div>
+    );
+};
+
+export const InputRadioNumber = (
+    {
+        id,
+        icon,
+        inputIconPath,
+        inputRef,
+        valueRange,
+        size,
+        overMaxAllowed
+    }: InputRadioNumberProps) => {
+
+    const [value, setValue] = useState(-1);
+    const [isOther, setIsOther] = useState(false);
+
+    const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setValue(event.target.valueAsNumber);
+    };
+
+    const choiceBuilder = (
+        idBase: string,
+        min: number,
+        max: number,
+        onChange: ChangeEventHandler<HTMLInputElement>,
+        icon?: IconProp,
+        inputIconPath?: { iconPath: string; iconSize: string; },
+    ) => {
+        const choiceList: ReactElement[] = [];
+
+        for (let i = min; i <= max; i+=1) {
+            choiceList.push(
+                <InputRadioNumberChoice
+                    key={`${idBase}_${i}`}
+                    id={`${idBase}_${i}`}
+                    value={i}
+                    icon={icon}
+                    inputIconPath={inputIconPath}
+                    onChange={handleOnChange}/>
+            );
+        }
+
+        return choiceList;
+    };
+
+    return (
+        <div className={'survey-question__input-radio-group-container'}>
+            {choiceBuilder(id, valueRange.min, valueRange.max, handleOnChange, icon, inputIconPath)}
+            {
+                overMaxAllowed &&
+                <>
+                    <div className="label-input-container">
+                        <input
+                            type="radio"
+                            id={'other'}
+                            className={'input-radio'}
+                            onChange={() => setIsOther(!isOther)}
+                        />
+                        <label htmlFor={'other'}>Other</label>
```

Comment:
The label should be max+, so 5+ if min: 1, max: 5 and overMaxAllowed: true, or 6+? I don't know, but as long as you document the behavior in the widgetconfig file, it's ok.